packages/hash/frontend/src/pages/[shortname]/types/entity-type/[entity-type-id].page/definition-tab/property-list-card/shared/property-type-form/expected-value-selector/custom-expected-value-builder/array-expected-value-builder/array-min-max-items.tsx

```diff
@@ -0,0 +1,302 @@
+import {
+  Box,
+  Checkbox,
+  Collapse,
+  collapseClasses,
+  Input,
+  inputClasses,
+  InputProps,
+  Typography,
+} from "@mui/material";
+import {
+  forwardRef,
+  FunctionComponent,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from "react";
+import { useController, useFormContext } from "react-hook-form";
+
+import { PropertyTypeFormValues } from "../../../../property-type-form-values";
+
+type ItemInputProps = { width: string } & InputProps;
+
+const ItemInput = forwardRef(({ width, ...props }: ItemInputProps, ref) => (
+  <Input
+    type="number"
+    {...props}
+    ref={ref}
+    sx={({ palette, transitions }) => ({
+      width,
+      fontSize: 11,
+      height: 16,
+      color: palette.white,
+      transition: transitions.create("color"),
+
+      "::before": {
+        display: "none",
+      },
+      "::after": {
+        borderBottomStyle: "dotted",
+        borderBottomColor: palette.white,
+        borderBottomWidth: 1,
+      },
+
+      ":hover": {
+        "::after": {
+          transform: "scaleX(1) translateX(0)",
+        },
+      },
+
+      [`.${inputClasses.disabled}`]: {
+        WebkitTextFillColor: "initial",
+      },
+
+      // <-- Hide number input default arrows -->
+      "& input[type=number]": {
+        WebkitAppearance: "textfield",
+      },
+      "& input[type=number]::-webkit-outer-spin-button": {
+        WebkitAppearance: "none",
+        margin: 0,
+      },
+      "& input[type=number]::-webkit-inner-spin-button": {
+        WebkitAppearance: "none",
+        margin: 0,
+      },
+      // <-- Hide number input default arrows -->
+    })}
+  />
+));
+
+interface ArrayMinMaxItemsProps {
+  arrayId: string;
+}
+
+export const ArrayMinMaxItems: FunctionComponent<ArrayMinMaxItemsProps> = ({
+  arrayId,
+}) => {
+  const { control, setValue } = useFormContext<PropertyTypeFormValues>();
+
+  const minItemsController = useController({
+    control,
+    name: `flattenedCustomExpectedValueList.${arrayId}.data.minItems`,
+  });
+  const maxItemsController = useController({
+    control,
+    name: `flattenedCustomExpectedValueList.${arrayId}.data.maxItems`,
+  });
+  const infinityController = useController({
+    control,
+    name: `flattenedCustomExpectedValueList.${arrayId}.data.infinity`,
+  });
+
+  const [minItemsWidth, setMinItemsWidth] = useState(0);
+  const [maxItemsWidth, setMaxItemsWidth] = useState(0);
+  const [hovered, setHovered] = useState(false);
```

Comment:
Nitpick: Maybe naming this `hoveredOnMax` / `maxHovered` (a name to indicate what is hovered) would be more self-explanatory. Because components is called min-max items. When I see `hovered`, I ask to myself `hovered to what?`. Totally optional ofc.

Reply:
I agree. Renamed in 800d4f659898b87cd231ea93104585886e861b24